### DISCUSSION TITLE
Purpose of Session module

### DISCUSSION THREAD

**Discussion by LucaAndrei** [brocoders/nestjs-boilerplate#1347] [Q&A] [answered]
What is the purpose of the session module?
Doesn't it defeat the purpose of stateless authentication with JWT when you are storing the session in the DB?
Are you storing the session to invalidate a user's session or to be able to blacklist a refresh token?
Thanks

**Comment by Shchepotin** [comment, marked as answer]
@LucaAndrei 

> What is the purpose of the session module?

Session module needs for the Refresh Token flow

> Doesn't it defeat the purpose of stateless authentication with JWT when you are storing the session in the DB?

Access tokens are still stateless

> Are you storing the session to invalidate a user's session or to be able to blacklist a refresh token?

For now "to be able to blacklist a refresh token"

**Comment by moris-moze** [comment]
@Shchepotin What's then the purpose of storing `sessionId` in the JWT (not refresh JWT) when signing it?

Code:

```
await this.jwtService.signAsync(
  {
    id: data.id,
    role: data.role,
    sessionId: data.sessionId,
  },
  {
    secret: this.configService.getOrThrow('auth.secret', { infer: true }),
    expiresIn: tokenExpiresIn,
  },
),
```

**Reply by Shchepotin** [reply]
@moris-moze for logout flow

https://github.com/brocoders/nestjs-boilerplate/blob/aeb15ffd9ddf9cb7d38f8699a38111c20f0410a7/src/auth/auth.controller.ts#L101-L109
